src/consensus/aft/raft.h

```diff
@@ -1128,7 +1128,7 @@ namespace aft
           apply_success = ds->execute();
           if (apply_success == kv::ApplySuccess::FAILED)
           {
-            rollback(last_committable_index());
+            state->last_idx = i - 1;
```

Comment:
Why not rollback here, what's the distinction?

Reply:
I believe there's no need for this path anymore: simply move `state->last_idx = i;` after this check and only return a failed append entries response (no need for `ledger->truncate(...)` either).

Reply:
As discussed I added a comment what is going here